blockstack/blockstack-browser
# Renew Blockstack IDs

This PR adds name expiry information and basic name renewal feature. Full on-chain names that have less than 60 days until expiration will see a renew button beside it. The renewal transaction will be funded by the built-in BTC wallet.

This is based on a [previous PR](https://github.com/blockstack/blockstack-browser/pull/1924) from @friedger to display expiry information in the all profiles screen.

Renew UI in all profiles screen
![image](https://user-images.githubusercontent.com/29081231/64041407-62056b80-cb2d-11e9-9c0b-ed5d888ccf18.png)

![image](https://user-images.githubusercontent.com/29081231/64041475-8f521980-cb2d-11e9-8e64-9be55d9cf416.png)

Subdomains do not need to be renewed
![image](https://user-images.githubusercontent.com/29081231/68799739-b1384400-0626-11ea-9672-56c299e11d7d.png)

## Testing notes
1. Find a name that has less than 60 days until expiration.
2. Go to the all profiles screen.
3. Click renew.
4. Enter password and press renew.
5. Wait for transaction to confirm x 10.
6. Verify that domain name now has many more days until expiration.
